### PR TITLE
Support subtract asset quantity command

### PR DESCRIPTION
**Pull request by MizukiSonoko**
## What is this pull request?
Re PR
https://github.com/hyperledger/iroha/pull/742

### CODE REVIEW

File: test/module/irohad/model/command_validate_execute_test.cpp [hunk 1]
```diff
@@ -212,6 +213,145 @@ TEST_F(AddAssetQuantityTest, InvalidWhenNoAsset) {
...
+    }
+
+    decltype(AccountAsset().balance) balance = Amount(150ul, 2);
+    Asset asset;
+    AccountAsset wallet;
+
+    std::shared_ptr<SubtractAssetQuantity> subtract_asset_quantity;
+};
+
+TEST_F(SubtractAssetQuantityTest, InvalidWhenNoWallet) {
```
Comment (l4l) [inline test/module/irohad/model/command_validate_execute_test.cpp]: We started using [gwt](https://en.wikipedia.org/wiki/Given-When-Then) way to write test docs, look at this for example:
https://github.com/hyperledger/iroha/blob/19651930fa6851586bfb9f09c3f6f2f3d3a51eaf/test/module/irohad/ametsuchi/block_query_test.cpp#L84-L90

File: test/module/irohad/model/operators/model_operators_test.cpp [hunk 2]
```diff
@@ -72,6 +73,31 @@ TEST(ModelOperatorTest, AddAssetQuantityTest) {
...
 }
 
+// -----|SubtractAssetQuantity|-----
+
+SubtractAssetQuantity createSubtractAssetQuantity() {
+  SubtractAssetQuantity saq;
+  saq.account_id = "acc";
+  iroha::Amount amount(1010, 2);
+  saq.amount = amount;
+  saq.asset_id = "ass";
```
Comment (grimadas) [inline test/module/irohad/model/operators/model_operators_test.cpp]: Better to use "ast" :D

File: irohad/model/execution/impl/command_executor.cpp [hunk 3]
```diff
@@ -261,6 +262,67 @@ namespace iroha {
...
+          log_->info("asset {} is absent", subtract_asset_quantity.asset_id);
+          return false;
+        }
+        auto precision = asset.value().precision;
+
+        if (subtract_asset_quantity.amount.getPrecision() != precision) {
+          log_->info("amount is wrongly formed");
+          return false;
+        }
+        if (not queries.getAccount(subtract_asset_quantity.account_id).has_value()) {
```
Comment (grimadas) [inline irohad/model/execution/impl/command_executor.cpp]: This query is not needed, you can remove it. If there is AccountAsset, it means the account exists

File: irohad/model/commands/subtract_asset_quantity.hpp [hunk 4]
```diff
@@ -17,8 +17,43 @@
...
+    namespace model {
+
+        /**
+         * Subtract amount of asset to an account
+         */
+        struct SubtractAssetQuantity : public Command {
+            /**
+             * Account where to subtract assets
+             */
+            std::string account_id{};
```
[the remark this answers is not in the document]
Comment (MizukiSonoko) [reply irohad/model/commands/subtract_asset_quantity.hpp]: How about other all commands? all command's string has this type constructor.

[the remark this answers is not in the document]
Comment (grimadas) [reply irohad/model/commands/subtract_asset_quantity.hpp]: In other commands they should removed too, if the command has a constructor.

[the remark this answers is not in the document]
Comment (MizukiSonoko) [reply irohad/model/commands/subtract_asset_quantity.hpp]: ok

**Pull request merged**